js/bitmex.js

```diff
@@ -1746,6 +1748,88 @@ module.exports = class bitmex extends Exchange {
         return this.parseTransaction (response, currency);
     }
 
+    async fetchFundingRateHistory (symbol = undefined, since = undefined, limit = undefined, params = {}) {
+        /**
+         * @method
+         * @name bitmex#fetchFundingRateHistory
+         * @description Fetches the history of funding rates
+         * @param {str} symbol Unified market symbol, use currency code to get data for the nearest expiring contract in that series, can also send a timeframe, eg XBT:quarterly, Timeframes are nearest, daily, weekly, monthly, quarterly, biquarterly, and perpetual
+         * @param {int} since timestamp in ms for starting date filter
+         * @param {int} limit number of results to fetch
+         * @param {dict} params exchange specific params
+         * @param {int} params.till timestamp in ms for ending date filter
+         * @param {bool} params.reverse if true, will sort results newest first
+         * @param {int} params.start starting point for results
+         * @param {str} params.columns array of column names to fetch in info, if omitted, will return all columns
+         * @param {str} params.filter generic table filter, send json key/value pairs, such as {"key": "value"}, you can key on individual fields, and do more advanced querying on timestamps, see the [timestamp docs]{@link https://www.bitmex.com/app/restAPI#Timestamp-Filters} for more details
+         * @returns A list of [funding rate history structures]{@link https://docs.ccxt.com/en/latest/manual.html#funding-rate-history-structure}
+         */
+        await this.loadMarkets ();
+        const request = {};
+        let market = undefined;
+        if (symbol in this.currencies) {
```

Comment:
I'm not sure if it's legit to lookup a trading pair (a symbol) in a dictionary of currencies in this case. Shouldn't we lookup a market and take its base or quote currency instead?

Reply:
The api accepts both, so if we don't accept both, then we're limiting the api a bit